Core/PrimaryGenerator.cpp

```diff
@@ -279,12 +293,12 @@ void PrimaryGenerator::GenerateEvtLargeChamber(G4Event* event)
   }
   else if (evtFractions[0] + evtFractions[1] + evtFractions[2] + evtFractions[3] > random) // Direct 3G
   {
-    event->AddPrimaryVertex(GenerateThreeGammaVertex(vtxPosition, T0, 
+    event->AddPrimaryVertex(GenerateThreeGammaVertex( MaterialExtension::DecayChannel::Direct, vtxPosition, T0, 
                             material->GetLifetime(random - evtFractions[0] - evtFractions[1] - evtFractions[2], MaterialExtension::DecayChannel::Direct)));
   }
   else // oPs 3G
   {
-    event->AddPrimaryVertex( GenerateThreeGammaVertex(vtxPosition, T0, 
+    event->AddPrimaryVertex( GenerateThreeGammaVertex( MaterialExtension::DecayChannel::Ortho3G, vtxPosition, T0, 
                             material->GetLifetime(random - evtFractions[1] - evtFractions[2] - evtFractions[3] - evtFractions[4], MaterialExtension::DecayChannel::Ortho3G)));
```

Comment:
random - evtFractions[0] - evtFractions[1] - evtFractions[2] - evtFractions[3]

Reply:
Could you also add additional else for pPs3G for LargeChamber as it is for small? it would be easier if we have the same type of generation for both chambers.